Sources/Request/Request/FormParam/Form.File.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  File.swift
+//  
+//
+//  Created by brennobemoura on 16/11/20.
+//
+
+import Foundation
+
+public extension Form {
+    struct File: FormParam {
+        private let path: Url
+        private let fileManager: FileManager
+        private let mime: String
+
+        public init(_ mediaType: MediaType, _ url: Url, _ fileManager: FileManager = .default) {
```

Comment:
I also think the `Url` should come first here. We can also likely infer the mime type from the file's `URL`:
```swift
guard let identifier = UTTypeCreatePreferredIdentifierForTag(kUTTagClassFilenameExtension, fileExtension, nil)?.takeUnretainedValue(),
      let mimeType = UTTypeCopyPreferredTagWithClass(identifier, kUTTagClassMIMEType)
else { throw SomeError }
```
Then `mediaType` (and `mime` below) could be optional. Also maybe include a label for `mediaType`:
```suggestion
        public init(_ url: Url, withType mediaType: MediaType? = nil, _ fileManager: FileManager = .default) throws {
```

Reply:
Nice, but I think that is better to separate in two different init, one that has the mimeType and the other that infer the mimeType